File: leetcode/python/1310.xor-queries-of-a-subarray.py
class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        xors = [0] * (len(arr) + 1)
        for i in range(len(arr)):
            xors[i+1] = xors[i] ^ arr[i]
        
        ans = [0] * len(queries)

        for i,q in enumerate(queries):
            ans[i] = xors[q[1] + 1] ^ xors[q[0]]
        return ans

File: leetcode/cpp/1310.xor-queries-of-a-subarray.cpp
class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
        vector<int> xors(arr.size()+1, 0);

        for (int i = 0; i < arr.size(); i++) {
            xors[i+1] = xors[i] ^ arr[i];
        }

        vector<int> ans(queries.size());

        for (int i = 0; i < queries.size(); i++) {
            ans[i] = xors[queries[i][1]+1] ^ xors[queries[i][0]];
        }
        return ans;
    }
};

